packages/mco/hooks/application-set.ts

```diff
@@ -0,0 +1,133 @@
+} from '../types';
+import {
+  findDeploymentClusterNameUsingPlacement,
+  findApplicationSetFromPlacement,
+  createRefFromApplicationSet,
+  isApplicationPresent,
+} from '../utils';
+
+const resources = (name?: string, namespace?: string) => ({
+  applications: {
+    name: name,
+    namespace: namespace,
+    kind: referenceForModel(ArgoApplicationSetModel),
+    isList: true,
+    namespaced: true,
+  },
+  placements: {
+    namespace: namespace,
+    kind: referenceForModel(ACMPlacementModel),
+    isList: true,
+    namespaced: true,
+  },
+  placementDecisions: {
+    namespace: namespace,
+    kind: referenceForModel(ACMPlacementDecisionModel),
+    isList: true,
+    namespaced: true,
+  },
+});
+
+export const useWatchApplicationSetResources: UseK8sWatchResource = (
+  resource
+) => {
+  const response = useK8sWatchResources<WatchResourceType>(
+    resources(resource?.name, resource?.namespace)
+  );
+  const memoizedResponse = useDeepCompareMemoize(response, true);
+
+  const {
+    data: placements,
+    loaded: placementsLoaded,
+    loadError: placementsLoadError,
+  } = memoizedResponse?.placements;
+
+  const {
+    data: placementDecisions,
+    loaded: placementDecisionsLoaded,
+    loadError: placementDecisionsLoadError,
+  } = memoizedResponse?.placementDecisions;
+
+  const {
+    data: applications,
+    loaded: applicationsLoaded,
+    loadError: applicationsLoadError,
+  } = memoizedResponse?.applications;
+
+  const loaded =
+    placementsLoaded && placementDecisionsLoaded && applicationsLoaded;
+
+  const loadError =
+    placementsLoadError || placementDecisionsLoadError || applicationsLoadError;
+
+  return React.useMemo(() => {
+    const acmApplicationPlacements: CustomAcmApplicationPlacementKind[] = [];
+    if (loaded && !loadError) {
+      placements.forEach((placement) => {
+        const usedApplications = findApplicationSetFromPlacement(
+          placement,
+          applications
+        );
+        // Ensure placement is under ramen control
+        const isFilterByAppName = !!resource?.name
+          ? isApplicationPresent(usedApplications, resource?.name)
+          : true;
+        if (usedApplications.length && isFilterByAppName) {
+          acmApplicationPlacements.push({
+            kind: placement?.kind,
+            name: getName(placement),
+            namespace: getNamespace(placement),
+            apiVersion: getAPIVersion(placement),
+            uid: getUID(placement),
+            deploymentClusterName: findDeploymentClusterNameUsingPlacement(
+              placement,
+              placementDecisions
+            ),
+            applicationRefs: createRefFromApplicationSet(usedApplications),
+          });
+        }
+      });
+    }
+    return [acmApplicationPlacements, loaded, loadError];
+  }, [
+    applications,
+    placements,
+    placementDecisions,
+    resource?.name,
+    loaded,
+    loadError,
+  ]);
+};
+
+type UseK8sWatchResource = (
```

Comment:
nits:
this is generic name, we should use specific name for our specific use case...
```suggestion
type UseK8sWatchResource = (
```